extensions/amp-next-page/1.0/service.js

```diff
@@ -161,9 +179,12 @@ export class NextPageService {
       this.setLastFetchedPage(this.hostPage_);
     }
 
-    this.parseAndQueuePages_();
+    this.nextSrc_ = this.getElement_().getAttribute('src');
+    this.hasDeepParsing_ =
+      this.getElement_().hasAttribute('deep-parsing') || !this.nextSrc_;
```

Comment:
@caroqliu open to suggestions on the name here, couldn't come up with something better. What it does is if enabled, it will also parse the list of suggested pages from the `<amp-next-page>` elements contained in the embedded documents (and adds them to the queue of suggested pages). See `Decision 1, Solution B` [here](https://docs.google.com/document/d/1tsCEo6UW5G6HmQn8eOvLNwuHKpfbxcWyxNgSIBuXThU/edit?pli=1#heading=h.1y810tw)

Reply:
Maybe `suggestDepthFirst`? I'll keep thinking about this.